sphinx/util/pycompat.py

```diff
@@ -79,11 +83,12 @@ def execfile_(filepath, _globals, open=open):
     try:
         code = compile(source, filepath_enc, 'exec')
     except SyntaxError:
-        if convert_with_2to3:
-            # maybe the file uses 2.x syntax; try to refactor to
-            # 3.x syntax using 2to3
-            source = convert_with_2to3(filepath)
-            code = compile(source, filepath_enc, 'exec')
-        else:
-            raise
+        # maybe the file uses 2.x syntax; try to refactor to
+        # 3.x syntax using 2to3
+        source = convert_with_2to3(filepath)
+        code = compile(source, filepath_enc, 'exec')
+        warnings.warn(
```

Comment:
Could you emit a warning for users instead using `logger.warning()` please?
It must be show a warning for users first before deprecating API.
We have to keep compatibility during 2.x (or 3.x).
So this helper function should not be deprecated during 2.x.

Reply:
Sure thing. That makes sense. For user facing warnings, should the feature be removed in 3.x or 4.x?

Reply:
Personally, removal in 3.x is a bit early. Because Sphinx-3.0 would be released at one year after 2.0 release. So I'd like to vote to removal in 4.0.